### PR TITLE
Variable word count and user-provided entropy

### LINKED ISSUE #2: Clean code & Add dynamic variables for user-supplied entropy and 24-word mnemonics
The current version of this simple Python program lets a user create a cryptographically secure 12-word mnemonic under BIP39. The issue posted here requires that the existing code is cleaned up (condensed, or otherwise improved for speed/performance) while adding the two options below:

1) User chooses at the start of the app whether to supply their own entropy or for the tool to randomly create entropy. 

2) User selects whether to use a 12-word (as it current defaults to) or a 24-word mnemonic (note: when users choose for the tool to randomly create the entropy, the option for word length will still be shown, and in all cases, the tool will also compute the check, and only allow 128 bits (not 132) for the user to add, as the tool should calculate those last 4 bits, or for 24 word mnemonics 256 bits (not 264) as the last 8 bit checksum would be calculated as part of validating the mnemonic.

### PR DESCRIPTION
**Pull request by Destiner**
* Clean up (PEP 8 complaint, split into functions, better naming)
* Ask user for word count (accepts 12 and 24, uses 12 as default)
* Ask user for entropy (accepts 0x-padded hex string, uses generated entropy as default)

Fixes #2.

### CODE REVIEW

File: bip39appv1_7.py [hunk 1]
```diff
@@ -37,14 +37,43 @@
 
 
 def main():
-    entropy = generate_entropy()
+    word_count = ask_word_count()
+    checksum_bit_count = word_count // 3
```
Comment (hatgit) [inline bip39appv1_7.py:41]: Why is this way (dividing wordcount by 3) of computing the length of the checksum better than by dividing the initial entropy by 32?

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Review by hatgit — APPROVED** [review]

**Comment by hatgit** [comment]
Great work @Destiner , I've looked through the code didn't have many questions. I will merge the commits now then run a few tests and roll-back commits if needed (perhaps best to set up a dev branch in the future).

File: bip39appv1_7.py (hunk 1 above)
Comment (Destiner) [reply bip39appv1_7.py:41]: Let me see. Basically, we don't know entropy from the start, right? What we have is word count, taken from the input. Is there any way to count entropy before counting number of bits in checksum? If there is, we can use it instead and then calc checksum bits by division to 32.

Comment (hatgit) [reply bip39appv1_7.py:41]: I think both options achieve the same result so not a big deal. Regarding your question, if a user chooses 12 then the resulting 128 bits that are either user-supplied or randomly generated could be divided by 32 to compute the checksum, I guess it's just another way to calculate but I am not seeing any difference in any performance gains either way. Thanks

File: bip39appv1_7.py [hunk 2]
```diff
@@ -37,14 +37,43 @@
...
-    entropy = format(secrets.randbits(130), '0129b')  # *get a minimum of x bits
-    entropy = entropy[:128]  # *use only bits required from the initial random number
+def ask_word_count():
+    default_word_count = 12
+    input_string = 'Enter word count: (12 or 24, default: {0}): '.format(default_word_count)
+    word_count_string = input(input_string)
+    if len(word_count_string) == 0:
+        return default_word_count
+    word_count = int(word_count_string)
+    while word_count != 12 and word_count != 24:
```
Comment (hatgit) [inline bip39appv1_7.py:57]: Hey @Destiner, tiny non-significant bug here when a user enters nothing in the first step (i.e. hits enter for default), then in the second step enters an invalid number (i.e. the digit "3") it gets stuck in the while loop and hitting the "enter" button at that stage  for the default doesn't work. Any suggestions on a quick fix? (not a big deal, thanks!)

Comment (Destiner) [reply bip39appv1_7.py:57]: Hey, sure, fixed in #5.

**Pull request merged**